src/transport_performance/gtfs/report/report_utils.py

```diff
@@ -0,0 +1,183 @@
+            self.template = f.read()
+        return None
+
+    def insert(
+        self, placeholder: str, value: str, replace_multiple: bool = False
+    ) -> None:
+        """Insert values into the html template.
+
+        Parameters
+        ----------
+        placeholder : str
+            The placeholder name in the template. This is a string. In the
+            template it should be surrounded by square brackets.
+        value : str
+            The value to place in the placeholder
+            location.
+        replace_multiple : bool, optional
+            Whether or not to replace multiple placeholders that share the same
+            placeholder value, by default False
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        ValueError
+            A ValueError is raised if there are multiple instances of a
+            place-holder but 'replace_multiple' is not True
+
+        """
+        _type_defence(placeholder, "placeholder", str)
+        _type_defence(value, "value", str)
+        _type_defence(replace_multiple, "replace_multiple", bool)
+        occurences = len(self.template.split(f"[{placeholder}]")) - 1
+        if occurences > 1 and not replace_multiple:
+            raise ValueError(
+                "You have selected not to replace multiple"
+                "placeholders of the same value, however"
+                "placeholders occur more than once. \n"
+                "If you would like to allow this, set the"
+                "replace_multiple param to True"
+            )
+
+        self.template = self.template.replace(f"[{placeholder}]", value)
+
+    def get_template(self) -> str:
+        """Get the template attribute of the TemplateHTML object.
+
+        Returns
+        -------
+        str
+            The template attribute
+
+        """
+        return self.template
+
+
+def set_up_report_dir(
+    path: Union[str, pathlib.Path] = "outputs", overwrite: bool = False
+) -> None:
+    """Set up the directory that will hold the report.
+
+    Parameters
+    ----------
+    path : Union[str, pathlib.Path], optional
+        The path to the directory,
+        by default "outputs"
+    overwrite : bool, optional
+        Whether or not to overwrite any current reports,
+        by default False
+
+    Returns
+    -------
+    None
+
+    Raises
+    ------
+    FileExistsError
+        Raises an error if you the gtfs report directory already exists in the
+        given path and overwrite=False
+
+    """
+    # defences
+    _check_parent_dir_exists(path, "path", create=True)
+
+    if os.path.exists(f"{path}/gtfs_report") and not overwrite:
+        raise FileExistsError(
+            "Report already exists at path: "
+            f"[{path}]."
+            "Consider setting overwrite=True"
+            "if you'd like to overwrite this."
+        )
+
+    # make gtfs_report dir
+    try:
+        os.mkdir(f"{path}/gtfs_report")
+    except FileExistsError:
```

Comment:
This line isn't covered by the test suite. Would this be a simple job to cover?

Reply:
I think this line would be covered if we simply ran through these lines once the gtfs_report had been created through testing

Reply:
Hi @r-leyshon. I've pushed a commit that addresses this lack of coverage